src/zenml/zen_stores/schemas/artifact_schemas.py

```diff
@@ -276,32 +276,34 @@ def to_model(self, hydrate: bool = False) -> ArtifactVersionResponse:
         Returns:
             The created `ArtifactVersionResponse`.
         """
+        try:
```

Comment:
In [the previous version of the artifacts](https://github.com/zenml-io/zenml/blob/6c266e3836163197b2f6181671c7d5bca445b381/src/zenml/models/artifact_models.py#L70), we were using a utility method called [convert_source_validator](https://github.com/zenml-io/zenml/blob/0807bf62bf0d2af4171917fe234a65ef14d21c5d/src/zenml/config/source.py#L203). Do you think that it is still applicable for the `ArtifactVersionResponseBody`?

Reply:
Nvm the message above, I just saw that we had it already in the `ArtifactVersionResponseBody` as a validator. Is there a reason why we have to do it here again?

Reply:
Theres a difference in what the two do: The `convert_source_validator` just converts strings to `Source` and I've moved that to the `ArtifactVersionResponseBody`. The code here loads a source from a dict.

Reply:
I see, this was a fun thread to resolve 😄  I was a bit confused because the `self.data_type` and `self.materializer` are `str`s by the definition of the schema. I guess they are serialized versions of the Source `object`.